lib/rspec/mocks/any_instance/proxy.rb

```diff
@@ -91,12 +91,18 @@ def perform_proxying(method_name, args, block, &target_proxy_block)
       # `targets` will typically contain 1 of the `AnyInstance::Recorder`
       # return values and N `MessageExpectation` instances (one per instance
       # of the `any_instance` klass).
-      class FluentInterfaceProxy
+      class FluentInterfaceProxy < defined?(BasicObject) ? BasicObject : Object
+        if superclass == ::Object
+          (instance_methods.map(&:to_sym) - [:__send__, :!, :instance_eval, :==, :instance_exec, :!=, :equal?, :__id__, :__binding__, :object_id]).each do |method|
```

Comment:
These methods came from running `BasicObject.instance_methods` in modern ruby and adding in `object_id` (as it being missing caused test failures).

Reply:
Can you add a comment explain the source of the methods please.